rootex/framework/systems/script_system.cpp

```diff
@@ -62,30 +62,58 @@ void CallUpdateForScene(Scene* scene, float deltaMilliseconds)
 void ScriptSystem::update(float deltaMilliseconds)
 {
 	ZoneScoped;
-
-	for (auto& entity : m_ScriptEntitiesToInit)
+	if (!(m_SystemPause))
 	{
-		if (entity)
+		for (auto& entity : m_ScriptEntitiesToInit)
 		{
-			entity->evaluateScriptOverrides();
-			entity->call("begin", { entity });
+			if (entity)
+			{
+				entity->evaluateScriptOverrides();
+				entity->call("begin", { entity });
+			}
 		}
-	}
-	m_ScriptEntitiesToInit.clear();
+		m_ScriptEntitiesToInit.clear();
 
-	for (auto& entity : m_ScriptEntitiesToEnter)
-	{
-		if (entity)
+		for (auto& entity : m_ScriptEntitiesToEnter)
 		{
-			entity->call("enterScene", { entity });
+			if (entity)
+			{
+				entity->call("enterScene", { entity });
+			}
 		}
+		m_ScriptEntitiesToEnter.clear();
+
+		Scene* root = SceneLoader::GetSingleton()->getRootScene();
+		CallUpdateForScene(root, deltaMilliseconds);
+
+		m_FluxTweener["update"](deltaMilliseconds * MS_TO_S);
```

Comment:
This part is common in both if and else. You can put it outside the conditional statements.